### PR TITLE
[url_launcher] Add iOS unit and UI tests

### PR DESCRIPTION
**Pull request by jmagman**
Add `RunnerTests` XCTest and `RunnerUITest` XCUITest iOS targets to example app.

- Add unit test that creates the plugin and asserts it's not nil, proving linking is set up correctly.
- Add UI test that launches different example web views.

url_launcher part of https://github.com/flutter/flutter/issues/83357

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`. See [plugin_tool format](../script/tool/README.md#format-code))
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/url_launcher/url_launcher/example/linux/flutter/generated_plugin_registrant.cc [hunk 1]
```diff
@@ -8,7 +8,6 @@
 
 void fl_register_plugins(FlPluginRegistry* registry) {
   g_autoptr(FlPluginRegistrar) url_launcher_linux_registrar =
-      fl_plugin_registry_get_registrar_for_plugin(registry,
-                                                  "UrlLauncherPlugin");
+      fl_plugin_registry_get_registrar_for_plugin(registry, "UrlLauncherPlugin");
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher/example/linux/flutter/generated_plugin_registrant.cc:11]: This file needs a revert. (Re-fixing this in Flutter so it stops happening is waiting to land.)

File: packages/url_launcher/url_launcher/example/ios/RunnerUITests/URLLauncherUITests.m [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  self.continueAfterFailure = NO;
+
+  self.app = [[XCUIApplication alloc] init];
+  [self.app launch];
+}
+
+- (void)testLaunch {
+  XCUIApplication* app = self.app;
+
+  NSArray<NSString*>* buttonNames = @[@"Launch in app", @"Launch in app(JavaScript ON)", @"Launch in app(DOM storage ON)", @"Launch a universal link in a native app, fallback to Safari.(Youtube)"];
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher/example/ios/RunnerUITests/URLLauncherUITests.m:24]: Interesting, I wonder if it's actually intentional that the last one falls back to SafariVC...

**Review by stuartmorgan — APPROVED** [review]
LGTM modulo making `format` happy.
